files/image_config/logrotate/rsyslog.j2

```diff
@@ -19,7 +19,9 @@
     delaycompress
     sharedscripts
     postrotate
-        /bin/kill -HUP $(cat /var/run/rsyslogd.pid)
+        if [ -f /var/run/rsyslogd.pid ]; then
+            /bin/kill -HUP $(cat /var/run/rsyslogd.pid)
```

Comment:
>kill [](http://example.com/codeflow?start=17&length=4)

Could you use `pkill -F`? no need to check pid file.

Reply:
I think that will still create an error in the case that the file does not exists.

```
$ pkill -F /var/run/rsyslogd.pid
pkill: pidfile not valid
Try `pkill --help' for more information.
$ echo $?
1
```

Reply:
@qiluo-msft 
I think that will still create an error in the case that the file does not exists.

```
$ pkill -F /var/run/rsyslogd.pid
pkill: pidfile not valid
Try `pkill --help' for more information.
$ echo $?
1
```